frontend/views/components/Modal/Modal.vue

```diff
@@ -15,15 +15,32 @@ export default {
   created () {
     sbp('okTurtles.events/on', LOAD_MODAL, component => this.openModal(component))
     sbp('okTurtles.events/on', CLOSE_MODAL, component => this.closeModal(component))
+    if (location.href.indexOf('?modal') > 0) {
+      const url = new URL(location.href)
+      this.openModal(url.searchParams.get('modal'), false)
+    }
   },
   beforeDestroy () {
     sbp('okTurtles.events/off', LOAD_MODAL, this.openModal)
     sbp('okTurtles.events/off', CLOSE_MODAL, this.closeModal)
   },
   methods: {
-    openModal (componentName) {
+    openModal (componentName, pushState = true) {
       this.content = componentName
       sbp('okTurtles.events/emit', OPEN_MODAL)
+      if (pushState) {
+        history.pushState(null, null, location.href + '?modal=' + componentName)
+      }
+      window.localStorage.setItem('modal', true)
+
+      // Update the URL to allow back button to close the popup
+      window.addEventListener('popstate', (event) => {
+        if (window.localStorage.getItem('modal')) {
+          window.localStorage.setItem('modal', false)
+          sbp('okTurtles.events/emit', CLOSE_MODAL)
+        }
+        window.removeEventListener('popstate', null, null)
```

Comment:
See [comment below](https://github.com/okTurtles/group-income-simple/pull/543/files?file-filters%5B%5D=.js&file-filters%5B%5D=.vue#r267117137)

Reply:
Good point I'll use the store.